emmet-core/emmet/core/qc_tasks.py

```diff
@@ -0,0 +1,609 @@
+        None,
+        description="Dictionary with the implicit solvent model and respective parameters",
+    )
+
+    lot_solvent: str = Field(
+        None,
+        description="A string representation with the combined information of the solvemt used and the level of theory",
+    )
+
+    custom_smd: str = Field(
+        None, description="Parameter string for SMD implicit solvent model"
+    )
+
+    special_run_type: str = Field(
+        None, description="Special workflow name (if applicable)"
+    )
+
+    smiles: str = Field(
+        None,
+        description="Simplified molecular-input line-entry system (SMILES) string for the molecule involved "
+        "in this calculation.",
+    )
+
+    calc_type: CalcType = Field(
+        None,
+        description="A combined dictionary representation of the task type along with the level of theory used",
+    )
+
+    @classmethod
+    def from_qchem_calc_doc(cls, calc_doc: Calculation) -> "InputDoc":
+        """
+        Create qchem calculation input summary from a qchem calculation document.
+
+        Parameters
+        ----------
+        calc_doc
+            A QChem calculation document.
+
+        Returns
+        --------
+        InputDoc
+            A summary of the input molecule and corresponding calculation parameters
+        """
+        # TODO : modify this to get the different variables from the task doc.
+        return cls(
+            initial_molecule=calc_doc.input.initial_molecule,
+            rem=calc_doc.input.rem,
+            # lev_theory = level_of_theory(calc_doc.input.rem),
+            task_type=calc_doc.task_type,
+            tags=calc_doc.input.tags,
+            solvent=calc_doc.input.solvent,
+            # lot_solvent_string = calc_doc.input.lot_solvent_string,
+            # custom_smd = calc_doc.input.custom_smd,
+            # special_run_type = calc_doc.input.special_run_type,
+            # smiles = calc_doc.input.smiles,
+            calc_spec=calc_doc.calc_type,
+        )
+
+
+class CustodianDoc(BaseModel):
+    corrections: List[Any] = Field(
+        None,
+        title="Custodian Corrections",
+        description="List of custodian correction data for calculation.",
+    )
+
+    job: dict = Field(
+        None,
+        title="Custodian Job Data",
+        description="Job data logged by custodian.",
+    )
+
+
+# AnalysisDoc? Is there a scope for AnalysisDoc in QChem?
+
+
+class TaskDoc(MoleculeMetadata):
+    """
+    Calculation-level details about QChem calculations that would eventually take over the TaskDocument implementation
+    """
+
+    dir_name: str = Field(None, description="The directory for this QChem task")
+
+    state: QChemStatus = Field(None, description="State of this QChem calculation")
+
+    calcs_reversed: List[Calculation] = Field(
+        None,
+        title="Calcs reversed data",
+        description="Detailed data for each QChem calculation contributing to the task document.",
+    )
```

Comment:
Okay, here we go. Nice. But if Calculation is meant to represent the full data in calcs_reversed, then that means that right now, you're throwing out a TON of information. Calculation will need to be seriously expanded. There's a reason that I previously just used dicts for calcs_reversed (other than me being lazy): there's just a lot of fields.